tof/models.py

```diff
@@ -98,26 +75,69 @@ class Meta:
 
     name = models.CharField(_('Field name'), max_length=250, help_text=_('Name field'))
     title = models.CharField(_('User field name'), max_length=250, help_text=_("Name user's field"))
-    content_type = models.ForeignKey(ContentType, limit_choices_to=~Q(app_label='tof'), on_delete=models.CASCADE)
+    content_type = models.ForeignKey(ContentType, limit_choices_to=~Q(app_label='tof'), on_delete=models.CASCADE, related_name='translatablefields')
 
     def __str__(self):
         return f'{self.content_type.model}|{self.title}'
 
     def save(self, *args, **kwargs):
         super().save(*args, **kwargs)
-        prepare_cls_for_translate(self.content_type.model_class(), self)
+        self.add_translation_to_class()
 
     def delete(self, *args, **kwargs):
-        cls = self.content_type.model_class()
-        ct_pk = self.content_type.pk
-        name = self.name
         super().delete(*args, **kwargs)
+        self.remove_translation_from_class()
+
+    def __get__(self, instance, instance_cls):
+        return instance.get_translation(self.name) 
+
+    def __set__(self, instance, value):
+        translation = vars(instance)[self.name] = instance.get_translation(self.name)  
+        setattr(translation, translation.get_lang() if hasattr(instance, '_end_init') else '_origin', str(value))
 
-        restore_cls_after_translate(
-            cls,
-            name,
-            ContentType.objects.filter(translatablefield__isnull=False, id=ct_pk).exists(),
-        )
+    def __delete__(self, instance):
+        vars(self).pop(self.name, None)
+        del type(instance)._meta._field_tof[self.attname]  # pragma: no cover
+
+    def save_translation(self, instance):
+        val = instance.get_translation(self.name)
+        if val:
+            translation, _ = self.translations.get_or_create(object_id=instance.pk, lang_id=val.get_lang(),)
```

Comment:
It do not work. Need pass content_type_id=self.content_type_id

Reply:
fixed